### PR TITLE
Delete loop from RooCintUtils::isValidEnumValue

### PR DESCRIPTION
**Pull request by yamaguchi1024**
We want to compare the first token of typeName separated by ":" and
value, and we only have to see it once.

### CODE REVIEW

File: roofit/roofitcore/src/RooCintUtils.cxx [hunk 1]
```diff
@@ -122,21 +123,15 @@ namespace RooCintUtils
...
-	}
-      }
-    }
-    gInterpreter->DataMemberInfo_Delete(dm);
-    gInterpreter->ClassInfo_Delete(cls);
-    return kFALSE ;
+
+    Bool_t res;
+
+    if (string(value)==gInterpreter->DataMemberInfo_Name(dm))
```
Comment (vgvassilev) [inline roofit/roofitcore/src/RooCintUtils.cxx]: What about just `return gInterpreter->DataMemberInfo_Name(dm) == string(value);`.

Comment (yamaguchi1024) [reply roofit/roofitcore/src/RooCintUtils.cxx]: Sounds good

Comment (yamaguchi1024) [reply roofit/roofitcore/src/RooCintUtils.cxx]: Is it Ok if we return bool not Bool_t?

Comment (pcanal) [reply roofit/roofitcore/src/RooCintUtils.cxx]: > Is it Ok if we return bool not Bool_t?

They are the same type :)

Comment (pcanal) [reply roofit/roofitcore/src/RooCintUtils.cxx]: > What about just return gInterpreter->DataMemberInfo_Name(dm) == string(value);.

Actually no ... at least not that simply as the code is missing:
```
  gInterpreter->DataMemberInfo_Delete(dm);
  gInterpreter->ClassInfo_Delete(cls);
```
which must be called before returning.

Comment (vgvassilev) [reply roofit/roofitcore/src/RooCintUtils.cxx]: Maybe in general we should introduce RAII helpers for these.

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by dpiparo** [comment]
nice PR! A very general question: Why is the class is called RooCintUtils? Maybe another name is more appropriate: RooInterpreterUtils?

File: roofit/roofitcore/src/RooCintUtils.cxx (hunk 1 above)
Comment (pcanal) [reply roofit/roofitcore/src/RooCintUtils.cxx]: This is a good idea.  We can even leverage the 'only-forward-declared-but-distinct' types (See TDictionary.h) to help out.   All we need is a way to associate the type with the factory function (i.e. ClassInfo_t to ClassInfo_Factory and ClassInfo_Delete) .... maybe we can use unique_ptr with a custom deleter as the RAII ..>)

**Comment by vgvassilev** [comment]
It is likely we can drop it because it seems almost unused, @yamaguchi1024 is looking into it.

**Comment by yamaguchi1024** [comment]
Add delete. Thanks for comments @vgvassilev @pcanal @dpiparo !

**Comment by yamaguchi1024** [comment]
@vgvassilev Yeah let's do it in next PR

**Comment by yamaguchi1024** [comment]
@dpiparo 
https://gist.github.com/yamaguchi1024/91d0605d7e2c730538101ececd427848
Like this? I think this is shorter and good!

**Comment by yamaguchi1024** [comment]
We don't need this anymore as #1722 merged

**Pull request closed without merging**